plugin/storage/es/spanstore/writer.go

```diff
@@ -53,22 +55,43 @@ type SpanWriter struct {
 
 // SpanWriterParams holds constructor parameters for NewSpanWriter
 type SpanWriterParams struct {
-	Client              es.Client
-	Logger              *zap.Logger
-	MetricsFactory      metrics.Factory
-	IndexPrefix         string
-	IndexDateLayout     string
-	AllTagsAsFields     bool
-	TagKeysAsFields     []string
-	TagDotReplacement   string
-	Archive             bool
-	UseReadWriteAliases bool
+	Client               es.Client
+	Logger               *zap.Logger
+	MetricsFactory       metrics.Factory
+	IndexPrefix          string
+	IndexDateLayout      string
+	AllTagsAsFields      bool
+	TagKeysAsFields      []string
+	TagDotReplacement    string
+	Archive              bool
+	UseReadWriteAliases  bool
+	ServiceCacheDuration string
+	IndexCacheDuration   string
```

Comment:
```suggestion
	ServiceCacheDuration time.Duration
	IndexCacheDuration   time.Duration
```

Reply:
You cannot check if a time.Duration is set or not since it would be 0, but you can check if a string is empty which is why I went the path I did there.

Anybody directly using the writer in their own code can use it and if later somebody wants to plumb in the ability for a flag for the apps to be able to set these could do that in a separate PR

I'm not sure how to test this since everything is in a private property and I cannot check if they are set in a test

Reply:
What's wrong with 0?  0 is not a valid value for these parameters, so it works perfectly well as "unset" value.

Tests run in the private package namespace and can test those values.

Reply:
0 is a valid value and would indicate caching forever so you still would not be able to differentiate from cache forever and not being set. Unless you want to assume that nobody would ever want to cache forever(which could be a valid thing, but I would rather be able to know for sure)

I did figure out how to test and just pushed those

Reply:
> 0 is a valid value and would indicate caching forever

Technically for the LRU cache that is under under the hood the ttl=0 is a valid param. But in the context of the specific caching scenarios here (index and service names) I do not believe 0 is a valid value, because these caches grow over time. So we can use 0 as "undefined". If people want to cache "forever" they can still do it by providing a very long duration.

Reply:
Sounds good. I can make the change

Reply:
If you really want to support "undefined" then the other alternative is to use a pointer.